scanpy/preprocessing/_pca.py

```diff
@@ -170,6 +170,10 @@ def pca(
         )
     data_is_AnnData = isinstance(data, AnnData)
     if data_is_AnnData:
+        if layer is None and not chunked and is_backed_type(data.X):
+            raise NotImplementedError(
+                f"PCA is not implemented for matrices of type {type(data.X)} from layers with chunked as False"
+            )
```

Comment:
hm would be nice to have the checks in the same place, but erroring early before doing work is probably more important.